File: src/Undersoft.SharedServiceCenter.Web.Application.React/src/undersoft.ssc.web.application.client/src/app/models/location.ts
import { Address } from "./address";
import { Endpoint } from "./endpoint";
import { Model } from "@/models/model";

export interface Location extends Model {
    name: string | null;
    localeType: LocaleType;
    email: string | null;
    phoneType: PhoneType;
    phoneNumber: string | null;
    notices: string | null;
    addresses: Address[];
    endpoints: Endpoint[] | null;
}

export enum LocaleType {
    Main,
    Private,
    Bussines,
    Additional
}

export enum PhoneType {
    Main,
    Personal,
    Bussines,
    Fax
}

File: src/SharedServiceCenter/src/Undersoft.SSC/Domain/Entities/Location.cs
using Undersoft.SDK.Service.Infrastructure.Store.Relation;
using Undersoft.SSC.Domain.Entities.Enums;
using Undersoft.SSC.Domain.Entities.Locations;

namespace Undersoft.SSC.Domain.Entities
{
    public class Location : DataObject
    {
        public virtual string? Name { get; set; }

        public virtual LocaleType LocaleType { get; set; }

        public virtual string? Email { get; set; }

        public virtual PhoneType PhoneType { get; set; }

        public virtual string? PhoneNumber { get; set; }

        public virtual string? Notices { get; set; }

        public virtual RelatedSet<Address>? Addresses { get; set; }

        public virtual RelatedSet<Endpoint>? Endpoints { get; set; }

        public virtual RelatedSet<Position>? Positions { get; set; }

        public virtual long? MemberId { get; set; }
        public virtual Member? Member { get; set; }

        public virtual long? PlatformId { get; set; }
        public virtual Platform? Platform { get; set; }

        public virtual long? ActivityId { get; set; }
        public virtual Activity? Activity { get; set; }

        public virtual long? ResourceId { get; set; }
        public virtual Resource? Resource { get; set; }

        public virtual long? ScheduleId { get; set; }
        public virtual Schedule? Schedule { get; set; }

        public virtual long? ServiceId { get; set; }
        public virtual Service? Service { get; set; }

        public virtual long? ApplicationId { get; set; }
        public virtual Application? Application { get; set; }

        public virtual long? EquipmentId { get; set; }
        public virtual Equipment? Equipment { get; set; }

        public virtual long? InfrastructureId { get; set; }
        public virtual Infrastructure? Infrastructure { get; set; }
    }
}
